src/components/course-periods-nav.cjsx

```diff
@@ -34,13 +34,15 @@ CoursePeriodsNav = React.createClass
     periods = CourseStore.getPeriods(courseId)
 
     period = periods?[key]
-    console.warn("#{key} period does not exist for course #{courseId}. There are only #{periods.length}.") unless period?
+    unless period?
+      console.warn("#{key} period does not exist for course #{courseId}. There are only #{periods.length}.")
```

Comment:
Can you `throw new Error('BUG: ...`? I'm hoping to replace all throw's with making an AJAX call to BE for debugging